### PR TITLE
Consolidate getQuota and setQuota methods in User instance

### PR DESCRIPTION
**Pull request by blizzz**
As @DeepDiver1975 did it with the email setter, we also should have the methods of reading and setting the quota within the user instance.

Please review @PVince81 @DeepDiver1975 @icewind1991

### CODE REVIEW

File: lib/private/user/user.php [hunk 1]
```diff
@@ -339,6 +340,36 @@ public function getEMailAddress() {
...
+	 * set the users' quota
+	 *
+	 * @param string $quota
+	 * @return void
+	 * @since 9.0.0
+	 */
+	public function setQuota($quota) {
+		if($quota !== 'none' and $quota !== 'default') {
+			$quota= OC_Helper::computerFileSize($quota);
+			$quota=OC_Helper::humanFileSize($quota);
```
Comment (PVince81) [inline lib/private/user/user.php]: this code needs some space :smile:

File: lib/public/iuser.php [hunk 2]
```diff
@@ -178,4 +178,21 @@ public function getCloudId();
...
+	/**
+	 * get the users' quota
+	 *
+	 * @return string
+	 * @since 9.0.0
+	 */
+	public function getQuota();
+
+	/**
+	 * set the users' quota
```
Comment (PVince81) [inline lib/public/iuser.php]: would be nice to have a hint about the format of that quota string, or a few examples

**Comment by PVince81** [comment]
Great stuff, setting quota this way looks much nicer.

**Comment by PVince81** [comment]
Looks like this still works, tested getting/setting from the users page and also querying the quota with Webdav.

:+1:

File: lib/private/user/user.php (hunk 1 above)
Comment (blizzz) [inline lib/private/user/user.php]: IDE should reformat on copy and paste ;D

**Comment by icewind1991** [comment]
:+1: looks good

**Comment by PVince81** [comment]
First error:
```
11:39:37 1) AvatarTest::testSetAvatar
11:39:37 Missing argument 1 for Mock_User_f009cec1::triggerChange(), called in /ssd/jenkins/workspace/server-master-linux-php7-ci/database/sqlite/label/SLAVE/lib/private/avatar.php on line 140 and defined
11:39:37 
11:39:37 /ssd/jenkins/workspace/server-master-linux-php7-ci/database/sqlite/label/SLAVE/lib/private/avatar.php:140
11:39:37 /ssd/jenkins/workspace/server-master-linux-php7-ci/database/sqlite/label/SLAVE/lib/private/avatar.php:122
11:39:37 /ssd/jenkins/workspace/server-master-linux-php7-ci/database/sqlite/label/SLAVE/tests/lib/avatartest.php:169
```

File: lib/public/iuser.php (hunk 2 above)
Comment (blizzz) [inline lib/public/iuser.php]: done

**Comment by PVince81** [comment]
:+1: for the last commits

File: apps/files_versions/lib/storage.php [hunk 3]
```diff
@@ -653,11 +653,9 @@ public static function expire($filename) {
 			$versionsFileview = new \OC\Files\View('/'.$uid.'/files_versions');
 
 			// get available disk space for user
+			$user = \OC::$server->getUserManager()->get($uid);
```
Comment (LukasReschke) [inline apps/files_versions/lib/storage.php]: :bomb: 

https://github.com/owncloud/core/issues/23513

File: apps/files_trashbin/lib/trashbin.php [hunk 4]
```diff
@@ -564,11 +564,8 @@ private static function calculateFreeSpace($trashbinSize, $user) {
 		$config = \OC::$server->getConfig();
 
 		$softQuota = true;
-		$quota = $config->getUserValue($user, 'files', 'quota', null);
+		$quota = \OC::$server->getUserManager()->get($user)->getQuota();
```
Comment (LukasReschke) [inline apps/files_trashbin/lib/trashbin.php]: This probably also needs some more error handling with regard to https://github.com/owncloud/core/issues/23513

File: apps/files_versions/lib/storage.php (hunk 3 above)
Comment (blizzz) [inline apps/files_versions/lib/storage.php]: I be damned.

**Pull request merged**